src/compiler/checker.ts

```diff
@@ -11512,6 +11513,32 @@ namespace ts {
                 }
             }
 
+            function findMostOverlappyType(source: Type, unionTarget: UnionOrIntersectionType) {

```

Comment:
OverlappyType is a new favorite term.

Reply:
Another candidate was `getMostShallowSatisfiedType` but that sounded slightly judgemental.

Reply:
So these types are always sorted by their ID, and those are always based on declaration order.

For named types (e.g. interface declarations) that's not usually that useful, but sometimes there's an intent when the user writes out anonymous types. For example, if the user wrote type literals such as `{ someProp: number } | { anotherProp: string }`, my own experience has shown me that a lot of the time the first one actually *is* thought of as the more common case. That's part of why I went with the first type in #26895 (though I think keeping it simple was the primary motivator).

When I tried this and only took the first most-overlappy type, I think I found a decent number of hit-or-miss changes. I didn't see significant improvements in the baselines and given that the change is literally switching between a `>` and a `>=`, I'd rather keep the baselines cleaner in that point. I think the high order bit is that we just look for overlappy types in the first place.

Reply:
Right, in our experience it ended up as more of a lavash or a naan, but I wouldn't be surprised if not sorting the IDs made type comparisons a pita.

Reply:
Barbari or Sangak 4ever.

Reply:
Please help @JsonFreeman

Reply:
I live in NYC now so I have to vote for bagels